geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClear.java

```diff
@@ -367,12 +370,38 @@ void doClear(RegionEventImpl regionEvent, boolean cacheWrite) {
           releaseLockForClear(regionEvent);
         }
       }
-
     } finally {
       releaseDistributedClearLock(lockName);
+      incClearDuration(System.nanoTime() - clearStartTime);
+    }
+  }
+
+  void incClearCount() {
+    if (partitionedRegion != null && partitionedRegion.getDataStore() != null
+        && partitionedRegion.getDataStore().getAllLocalBucketRegions() != null
+        && partitionedRegion.getDataStore().getAllLocalBucketRegions().size() != 0) {
```

Comment:
I'm not sure that all these null checks are required. I would think that if the region is null at this point, then we would expect that something has gone wrong and that we should throw an exception rather than just silently not incrementing the stats. Also, I'm not sure I understand why we only increment the stats if `getAllLocalBucketRegions()` returns a non-empty set. Are there scenarios where we could call clear, it could complete without an exception, bringing us to this method, and then `getAllLocalBucketRegions()` returns empty?

Reply:
Checking that the partitioned region is non null might be overkill, but the others are needed to determine whether or not this member has data. If there's no data on the member then no clear is actually taking place and we don't want to log a count. I added this many checks because I saw inconsistent behavior in my tests where sometimes the PR would have a list which was empty, sometimes it would have a list which was null for my Accessor. I don't think it's necessary to throw an exception in this case.

Reply:
Might it be possible to instead move the call to this method (and `incClearDuration()`, if it ends up staying where it currently is) inside the `if (partitionedRegion.getDataStore() != null)` block in `clearRegionLocal()`, just after the `doAfterClear()`, and guarded with a check on the size of `clearedBuckets`? i.e. if no buckets are cleared, do not increment the stats? That should avoid having to do quite so much checking in the method itself and instead place the focus on calling it in the appropriate place, where it will always increment the stats.

Reply:
It could, but that pattern would have to be followed if this clear count was ever incremented from anywhere else in the code base, as it would be reliant upon being called within a conditional. Is that preferable? Personally, I feel like having the conditions - while a little messy - communicates the requirements of the method pretty clearly and while I wouldn't necessarily promote that approach for a more complex method which is already very busy with functionality, this method is simple enough to read even with the conditions. Aside from that I think what you're suggesting is certainly possible.

Reply:
If all the checks were removed from the method, then it would turn into just 'partitionedRegion.getCachePerfStats().incRegionClearCount()` which could be called inline rather than having a separate method, especially considering that it's only called once. Looking at how other stats are incremented, this pattern of directly calling the method on the stats class inline when it's appropriate to do so seems to be used almost everywhere, and using it here would simplify the class a fair bit.